app/src/main/java/com/example/mona/facebookoffline/MainActivity.java

```diff
@@ -2,21 +2,57 @@
 
 import android.app.Activity;
 
+import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
+import android.widget.Toast;
 
 import com.example.mona.facebookoffline.auth.LoginFragment;
+import com.facebook.AccessToken;
+import com.facebook.FacebookException;
+import com.facebook.FacebookSdk;
+import com.facebook.login.LoginResult;
 
 /**
  * Main activity, allows a user to login, create new posts, and access old posts
  */
-public class MainActivity extends Activity {
+public class MainActivity extends Activity implements LoginFragment.LoginListener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        // Initialize FB sdk
+        FacebookSdk.sdkInitialize(this);
         setContentView(R.layout.activity_main);
+
+        Button create_button = (Button) findViewById(R.id.create_post);
+        create_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(MainActivity.this, EditActivity.class));
+            }
+        });
+    }
+
+    @Override
+    public void onSuccess(LoginResult loginResult) {
+        launchMainActivityAndFinish();
```

Comment:
This seems weird. It's basically killing the current MainActivity and launching a new one. I think the code will work fine if we just get rid of this line! Maybe replace it with a log statement :)